.gitlab-ci.yml

```diff
@@ -0,0 +1,29 @@
+before_script:
+  - docker info
+  # check the availability of some commands:
+  - command >/dev/null -v python3 jq
+
+build-in-docker:
+  stage: build
+  script:
+    - cd tools/docker
+    # Try to pull from dockerhub, build the container images if that fails
+    - ./image-pull.sh || ./image-build.sh
+    # Run clang-format and check there are no modified files by checking git status
+    - ./clang-format.sh
+    - git diff-index --exit-code HEAD
+    - ./build.sh map -f CMAKE_BUILD_TYPE=Debug BUILD_TESTS=ON
+    # Check that the AutoGenerated files are correct by checking git status
+    - git update-index -q --refresh
+    - git diff-index --exit-code HEAD
+  tags:
+    - shell
+
+run-tests:
+  stage: test
+  script:
+    - cd tools/docker
+    - ./run.sh -f -d --entrypoint "$PWD/../../../build/install/bin/tests/tlvf_test"
+    - ./tests/test_flows.sh -v
```

Comment:
This will leave the test containers running directly in the machine which has the runner (i.e., mind2). Not nice... Please add a after_script to kill the containers. This **is** something that must be fixed for this PR.

Also, since this is in a shell runner, the containers started by different instances all have the same name "gateway", "repeater1", "repeater2". Therefore, if two jobs are executing concurrently, they'll conflict with each other. We need to find a way to resolve that - but again, can be done later.